### PR TITLE
Ddc upgrade

### PR DESCRIPTION
**Pull request by devoncarew**
@financeCoding, I'm working on gettings dartdocs.org upgraded from `docgen` to `dartdoc`. You can see the new output of `dartdoc` here: https://dartdoc.firebaseapp.com/.

This PR:
- moves from `unittest` to `test`
- ignores new pub files
- removes some analysis warnings
- upgrades other packages to the latest
- get the tests running
- convert the call from `docgen` to `dartdoc`; remove the dartdoc viewer step

I'm not familiar enough with this project to know how to test or run it very well. Is there a way I can test the generation path w/ having access to any of the google storage buckets?

### CODE REVIEW

File: lib/src/package.dart [hunk 1]
```diff
@@ -139,66 +139,64 @@ class Package {
...
-    Directory outDartdocViewerDirectory = new Directory(outDartdocViewerPath);
-
-    outWebComponentsDirectory.renameSync(webWebComponentsPath);
-    outPolymerDirectory.renameSync(webPolymerPath);
-    outDartdocViewerDirectory.renameSync(webDartdocViewerPath);
-  }
+  // /**
+  //  * Moves the packages folder into the root of the web folder.
+  //  *
+  //  * WARNING: this may change in the future versions dartdoc-viewer.
```
Comment (financeCoding) [inline lib/src/package.dart]: Whats the current state of this? No longer needed folder juggling

File: lib/src/package.dart [hunk 2]
```diff
@@ -139,66 +139,64 @@ class Package {
...
    */
   int buildDocumentationSync(Version version, String dartSdkPath, {bool verbose: false}) {
     String outputFolder = 'docs';
-    String packagesFolder = './packages'; // The pub installed packages
+    //String packagesFolder = './packages'; // The pub installed packages
     String workingDirectory = join(BUILD_DOCUMENTATION_ROOT_PATH,
         "${name}-${version}");
 
-    List<String> args = ['--compile',
-                         '--no-include-sdk',
```
Comment (financeCoding) [inline lib/src/package.dart]: These are default now or linked differently to the root documentation? 

```
--no-include-sdk
--no-include-dependent-packages
```

File: pubspec.yaml [hunk 3]
```diff
@@ -5,20 +5,20 @@ authors:
...
-  meta: ">=0.8.8 <0.9.0"
-  path: ">=1.1.0 <2.0.0"
-  logging: ">=0.9.0 <0.10.0"
-  crypto: ">=0.9.0 <0.10.0"
-  mustache: ">=0.1.5 <0.2.0"
-  quiver: ">=0.18.2 <0.19.0"
-  unscripted: ">=0.4.2 <0.5.0"
-  uuid: ">=0.3.2 <0.3.3"
-  route: ">=0.4.6 <0.4.7"
+  args: any
```
Comment (financeCoding) [inline pubspec.yaml]: Is `any` the new suggested way to do versioning for packages? I'm scared of it cause of past experiences of breaking APIs

**Comment by financeCoding** [comment]
sgtm. I think after these changes the daemon would need to be redeployed, I don't see any reason why it might except for possibly datastore issues. I could possibly test tomorrow night. If your feeling adventurous try restarting the daemon before merging this with (might need to make sure your gcloud/gcutil auth is defaulted to the project)

```
cd dart-carte-du-jour/scripts/daemon_launch
source daemon-isolate-functions.sh
carte_restart_daemon_isolate
```

If that works and it seems to be generating packages, then go forward and merge this, pull master to local git repo and restart the daemon again.

File: pubspec.yaml (hunk 3 above)
Comment (devoncarew) [inline pubspec.yaml]: Sorry, I just did this to help me upgrade to the latest packages. Will change these to more restricted ranges.

File: lib/src/package.dart (hunk 1 above)
Comment (devoncarew) [inline lib/src/package.dart]: Yeah, the output from dartdoc isn't a dart app w/ `packages/` folders, but just raw html+css. No more separate docgen and dartdoc-viewer.

File: lib/src/package.dart (hunk 2 above)
Comment (devoncarew) [inline lib/src/package.dart]: dartdoc doesn't generate or link to the sdk (or other package source) by default. It _can_ optionally generate separate docs for the sdk.

**Comment by sethladd** [comment]
thanks!

**Pull request merged**